src/fl_draw.cxx

```diff
@@ -80,10 +80,12 @@ static const char* expand_text_(const char* from, char*& buf, int maxbuf, double
     if (o > e) {
       if (maxbuf) break; // don't overflow buffer
       l_local_buff += (o - e) + 200; // enlarge buffer
+      auto d_o = (o - local_buf);
+      auto d_word_end = (word_end - local_buf);
       buf = (char*)realloc(local_buf, l_local_buff);
       e = buf + l_local_buff - 4; // update pointers to buffer content
-      o = buf + (o - local_buf);
-      word_end = buf + (word_end - local_buf);
+      o = buf + d_o;
+      word_end = buf + d_word_end;
```

Comment:
Thanks for all your effort. I apologise for nagging again. For back compatibility, FLTK does not use "auto". IIRC `ptrdiff_t` from `stddef.h` is permitted.

Reply:
This has been fixed in 1.4.0 (using size_t BTW). If at all we should use the code from 1.4.0 for a fix.

Note: I explained in fltk.general that we don't accept patches for pure warnings. In this case I would say this can be an issue and would backport the 1.4.0 code.